server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -213,6 +213,22 @@ private synchronized boolean assertNoSeqNumberConflict(long seqNo, BytesReferenc
         return true;
     }
 
+    synchronized boolean assertNoSeqAbove(long belowTerm, long aboveSeqNo) throws IOException {
+        if (aboveSeqNo != SequenceNumbers.UNASSIGNED_SEQ_NO) {
+            TranslogSnapshot snapshot = newSnapshot();
+            Translog.Operation operation;
+            while ((operation = snapshot.next()) != null){
```

Comment:
why did you move away from the map based assertion you had before? I'm fine with this one as well, just curious to know.

Reply:
map contains only seq# - there is no any sematic guarantees on primary terms on adding operations - therefore i'd prefer to have a full check both on seq# and on primaryTerms